Improvements to staff entry user interface, and minor bug fixes.

misplacedpriorities opened the pull request:
(1) Added a modal dialog used to add or edit staff entries. These are triggered by buttons in the "sort by name" version of the hotline staff page: a button in each row allowing editing of existing staff entries, and a button at the bottom of the table allowing addition of new staff entries.

(2) Added CSV export to the bottom of the hotline staff page, allowing staff and call time entries to be exported in the same format as they are imported in the text field right above it.

(3) Fixed a bug that allowed the addition of call times via CSV import that included no 'receive type' (texts, calls, or alerts).

(4) Changed locking of tables to be more comprehensive when adding staff and/or call time entries via CSV import, or when deleting staff.

sharett commented on export.php:
```diff
@@ -34,76 +42,125 @@
+                echo $error;
+                break;
+            }
+            fwrite($fp, "name,number,day,earliest,latest,language_id,receive_texts,".
```
Not sure if this was intentional - the heading is written out for each contact even if the contact has no call times.

misplacedpriorities replied:
Good catch, fixed.

misplacedpriorities commented:
Good catch on that header being written for every contact; I believe that's been fixed.

I think there were some errors during what I'm guessing was an automatic merge of the latest master changes with this branch. Those should be fixed (please double-check the diff between my latest commit and the previous version for communications_export.php and export.php).

sharett commented:
Hi Chris - I figured it out.  You have to select "by name" in order to add anything.  When you are in the "by day" section, if there are no records, there's nothing to select.  Would it make sense to have a notice there that says to switch to "by name" to add records?

misplacedpriorities commented:
Good catch. I've added a row at the bottom of the "sort by day" table that allows the user to add a new staff entry, just as with the "sort by name" table. For symmetry's sake, I also added a button to each entry in the "by day" table to allow the user to edit that staff entry, just as with the other table.

sharett commented:
A couple more thoughts: When there are no call times, the "New staff entry" button doesn't show up - could that be present instead of or in addition to the "There are no call times on record."?

When adding a new staff entry while in the "by day" section, could it switch to the "by name" section, so that the newly added staff entry is visible?

Thanks!

misplacedpriorities commented:
Good suggestions (the first issue was actually a bug; the second one was performing as intended but I agree that it wasn't optimal). I've implemented both of these changes in the latest patchset.
